designs/nutanix-datacenterconfig-credentialref.md

```diff
@@ -0,0 +1,51 @@
+# Nutanix Datacenter Config Credential Reference
+
+## Introduction
+
+**Problem:** When a management cluster and a workload cluster are present on two different Nutanix Prism Central instances,
+using same credentials to authenticate to both instances is not possible. To remedy this, we need to associate credentials
+with corresponding Nutanix Datacenter Configs.
+
+## Overview of Solution
+With this feature, a user can specify credentials reference (i.e. Name and Namespace of the secret containing the credentials)
+in the Nutanix Datacenter Config. This info will be fetched and used to authenticate to the Nutanix Prism Central instance
+when creating the workload cluster.
+
+### Solution Details
+Extend the `NutanixDatacenterConfig` CRD to include a new field `credentialRef` storing name, namespace, and kind (similar to how
+`credentialsRef` is defined in the `NutanixCluster` CRD).
+
+```go
+// NutanixDatacenterConfigSpec defines the desired state of NutanixDatacenterConfig.
+type NutanixDatacenterConfigSpec struct {
+    ...
+    // CredentialRef is the reference to the secret that contains the credentials
+    // for the Nutanix Prism Central
+    // +optional
+    CredentialRef *NutanixCredentialReference `json:"credentialRef,omitempty"`
```

Comment:
Could we use a non specific nutanix type for this?
I believe we have already a ref (or more than one) types, I would love to reuse it.
If for whatever reason they don't serve you purpose, maybe we can go with `type CredentialsRef struct`?

Reply:
@jiayiwang7 suggests we use `Ref`. I'm game for common data structures, however, `Ref` seems to lack `Namespace` https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L741